### PR TITLE
Fixes #30360: Drop Pulp 2 and Qpid

### PR DESCRIPTION
**Pull request by ehelms**

### CODE REVIEW

**Comment by ehelms** [comment]
Related to https://github.com/theforeman/puppet-foreman_proxy_content/pull/271 but does not have the same level of re-factoring impact since all Pulpcore configuration exists over in puppet-foreman_proxy_content.

@jlsherrill can you look and make sure the `katello.yml` is accurate?

File: manifests/application.pp [hunk 1]
```diff
@@ -20,7 +20,6 @@
   Boolean $use_pulp_2_for_file = false,
   Boolean $use_pulp_2_for_docker = false,
   Boolean $use_pulp_2_for_yum = false,
```
Comment (ekohl) [inline manifests/application.pp:22]: These can be dropped as well if there's no Pulp 2.

File: manifests/application.pp [hunk 2]
```diff
@@ -20,7 +20,6 @@
...
-  Stdlib::Absolutepath $repo_export_dir = '/var/lib/pulp/katello-export',
```
Comment (ekohl) [inline manifests/application.pp:23]: Please also drop the docs for this one

File: templates/katello.yaml.erb [hunk 3]
```diff
@@ -7,10 +7,7 @@
   :content_types:
     :file: <%= @enable_file %>
     :yum: <%= @enable_yum %>
-    :deb: <%= @enable_deb %>
-    :puppet: <%= @enable_puppet %>
-    :docker: <%= @enable_docker %>
-    :ostree: <%= @enable_ostree %>
+    :docker: <%= @enable_container %>
```
Comment (ekohl) [inline templates/katello.yaml.erb:10]: AFAIK content types are Pulp 2 only. With Pulp 3 they're discovered via capabilities in the Smart Proxy

File: templates/katello.yaml.erb [hunk 4]
```diff
@@ -24,7 +21,6 @@
     :ssl_ca_file: <%= @candlepin_ca_cert %>
 
   :pulp:
-    :url: <%= @pulp_url %>
     :ca_cert_file: <%= @pulp_ca_cert %>
 
   :use_pulp_2_for_content_type:
```
Comment (ekohl) [inline templates/katello.yaml.erb:26]: Feels like this could be removed as well

File: manifests/application.pp (hunk 1 above)
Comment (ehelms) [reply manifests/application.pp:22]: I agree they ought to be dropped, just waiting for confirmation from @jlsherrill that they have been dropped code-wise inside Katello.

Comment (ekohl) [reply manifests/application.pp:22]: The code suggests that if it's not present, it equals to false so should be safe to drop:
https://github.com/Katello/katello/blob/cf1bf09002252669c997b6317868582d2c35b465/app/models/katello/concerns/smart_proxy_extensions.rb#L149-L156

Comment (jlsherrill) [reply manifests/application.pp:22]: yep +1 safe to drop

File: templates/katello.yaml.erb (hunk 3 above)
Comment (jlsherrill) [reply templates/katello.yaml.erb:10]: these are technically still needed as this tells katello what nav items to show, and what type of repositories can be created.  We could look into looking at the smart proxy for this, but aren't currently.

File: templates/katello.yaml.erb (hunk 4 above)
Comment (jlsherrill) [reply templates/katello.yaml.erb:26]: :+1:

File: templates/katello.yaml.erb [hunk 5]
```diff
@@ -33,8 +29,3 @@
     :yum: <%= @use_pulp_2_for_yum %>
 
   :katello_applicability: <%= !@use_pulp_2_for_yum %>
```
Comment (jlsherrill) [inline templates/katello.yaml.erb:31]: this could just be set to 'true' for now

File: templates/katello.yaml.erb (hunk 3 above)
Comment (ekohl) [reply templates/katello.yaml.erb:10]: Can we open an issue for that to track? From the installer side we want as little configuration as possible and a more dynamic application would be the result.

**Pull request closed without merging**